src/main/java/de/zalando/aruha/nakadi/exceptions/NakadiException.java

```diff
@@ -26,4 +31,11 @@ public void setProblemMessage(final String problemMessage) {
         this.problemMessage = problemMessage;
     }
 
+    public Problem asProblem() {
+        final String detail = StringUtils.isNotBlank(problemMessage) ? problemMessage : getMessage();
+        return Problem.valueOf(getStatus(), detail);
+    }
+
+    protected abstract Response.StatusType getStatus();
```

Comment:
Another approach would be to have a "problem factory" that receives a "domain exception" and returns a "problem".